ui/src/app/icons/edit/component.tsx

```diff
@@ -0,0 +1,77 @@
+import {Component} from '@angular/core';
+
+import * as React from 'react';
+
+import {ReactWrapperBase} from 'app/utils';
+
+interface EditComponentState { style: object; }
+
+const style = {
+  height: 14,
+  width: 14,
+};
+
+const fillStyle = {
+  fill: '#4D4CA5'
+};
+
+class EditComponentReact extends React.Component<{}, EditComponentState> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      style: style
```

Comment:
While we do need state to track the hover, we could minimize it a bit by just storing the hover state as a boolean, and computing the final style during render. (However, see below)

Reply:
Got it - yes, that makes sense.